docs/docs/components/appnav.md

```diff
@@ -0,0 +1,135 @@
+
+```java
+AppNavItem dashboardItem = new AppNavItem("Dashboard", "/dashboard");
+AppNavItem helpItem = new AppNavItem("Help", "https://support.example.com");
+
+appNav.addItem(dashboardItem);
+appNav.addItem(helpItem);
+```
+
+**Registered View Class**: alternatively, if you have views registered with your app’s router, you can pass a class reference for the view. This approach provides flexibility, as it allows navigation to specific views without hardcoding URLs. The router automatically resolves the view’s path based on the registered configuration.
+
+```java
+AppNavItem settingsItem = new AppNavItem("Settings", SettingsView.class);
+appNav.addItem(settingsItem);
+```
+
+:::tip Query Parameters
+To navigate to specific sections, use the `setQueryParameters()` method with a `ParametersBag` to define key-value pairs. 
+:::
+ 
+
+## Setting navigation targets
+
+The `AppNavItem` component provides the `setTarget()` method, which allows you to control the behavior of each navigation item when it's clicked. By default, items open in the current browsing context, but you can customize this with the `NavigationTarget` options:
+
+- **SELF**: This is the default option and opens in the current browsing context.
+- **BLANK**: Opens the item in a new tab or window based on browser settings.
+- **PARENT**: Opens in the parent browsing context; if there’s no parent, it behaves like SELF.
+- **TOP**: Opens in the top-level browsing context. If there’s no higher-level context, it also behaves like SELF.
+
+```java
+AppNavItem dashboardItem = new AppNavItem("Dashboard", "/dashboard");
+dashboardItem.setTarget(AppNavItem.NavigationTarget.SELF); // Opens in the current view
+
+AppNavItem helpItem = new AppNavItem("Help", "https://support.example.com");
+helpItem.setTarget(AppNavItem.NavigationTarget.BLANK); // Opens in a new tab
+
+AppNavItem parentItem = new AppNavItem("Parent View", "/parent");
+parentItem.setTarget(AppNavItem.NavigationTarget.PARENT); // Opens in the parent context
+
+AppNavItem topItem = new AppNavItem("Top View", "/top");
+topItem.setTarget(AppNavItem.NavigationTarget.TOP); // Opens in the top-level context
+```
+
+## Item prefix and suffix
+
+The `AppNavItem` component allows you to customize its appearance by adding prefixes and suffixes. These are useful for enhancing the visual design or providing additional context, such as icons, badges, or other indicators, alongside the item's text.
+
+- **Prefixes**: Add elements or icons to appear before the item's label. For example, you can use an icon to represent the item's purpose visually.
+
+- **Suffixes**: Add elements or icons to appear after the item's label. This is often used for secondary information, such as a badge showing a count or a status indicator.
+
+Here's an example of adding a prefix and suffix:
+
+```java
+AppNavItem notificationsItem = new AppNavItem("Notifications");
```

Comment:
These are just made-up snippets that don’t actually work. Please use real-world examples that are functional and can be used in practice.